homeassistant/components/broadlink/time.py

```diff
@@ -0,0 +1,63 @@
+"""Support for Broadlink device time."""
+
+from __future__ import annotations
+
+from datetime import time
+from typing import Any
+
+from homeassistant.components.time import TimeEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from . import BroadlinkDevice
+from .const import DOMAIN
+from .entity import BroadlinkEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Broadlink time."""
+    device = hass.data[DOMAIN].devices[config_entry.entry_id]
+    async_add_entities([BroadlinkTime(device)])
```

Comment:
Let's add some flow control to ensure that if the entity is of type 'HYS', then we add BroadlinkHysenTime entities. Otherwise, we do nothing.

Reply:
While I get the idea I think this would be different then how filtering is done for other platforms. E.g. https://github.com/home-assistant/core/blob/27a2114ba914af535713651c4c9a72cf31b255c1/homeassistant/components/broadlink/sensor.py#L95-L104

Reply:
Even if we do a good job creating a unified interface in the library, we could still have at least two types of time devices: weekday/time and date/time.

Reply:
Moreover we do already filter the platforms by device type: https://github.com/home-assistant/core/pull/121470/files#diff-de5887f91e2b6b4559fe8fe3c68b104fbbcc5048feb84a0db325be16527d87f0R39

Why should we do it again?

Reply:
It's about making the code more robust. This function accepts a `ConfigEntry` in its signature. It should be able to handle _any_ `ConfigEntry`, without depending on filtering or validation by a higher level module.